src/combinatorics/simple_graph/partition.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2021 Arthur Paulino. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Kyle Miller
+-/
+
+import combinatorics.simple_graph.coloring
+
+/-!
+# Graph partitions
+
+This module provides an interface for dealing with partitions on simple graphs. A partition of
+a graph `G`, with vertices `V`, is a set `P` of disjoint subsets of `V` such that:
+
+* The union of all elements of `P` resuls on `V`.
+
+* Each element of `P` doesn't contain a pair of vertices incident by the same edge;
+
+This module also explores the relationship between partitions and colorings. A partition can be
+created from a coloring by grouping vertices by their colors. Similarly, a coloring can be created
+from a partition by coloring every vertex in the same subset of `V` with the same color.
+
+## Main definitions
+
+* `partition` is a structure to represent a partition of a simple graph
+
+* `to_coloring` and `to_coloring'` are functions that create colorings from partitions
+
+* `from_coloring` is a function that creates a partition from a coloring
```

Comment:
Please add the namespaces to these, which will make the docgen turn them into links.